src/plugins/plugins_ex.h

```diff
@@ -386,119 +333,141 @@ struct breakpoint_by_pid_searcher
     }
 };
 
+template<typename Plugin, typename Params = void>
+struct plugin_data
+{
+    Plugin* plugin;
+    Params* params;
+
+    plugin_data(Plugin* plugin, Params* params)
+        : plugin(plugin), params(params)
+    {};
+};
+
 class pluginex : public plugin
 {
 public:
     typedef event_response_t(*hook_cb_t)(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
 
-public:
-    pluginex(drakvuf_t drakvuf, output_format_t output);
-    virtual ~pluginex();
+    pluginex(drakvuf_t drakvuf, output_format_t output)
+        : m_output_format(output), drakvuf(drakvuf)
+    {};
 
-public:
-    template<typename P = pluginex, typename Params = plugin_params<P>, typename IB,
-             typename AP = allocator<Params>, typename AT = allocator<drakvuf_trap_t>>
-    drakvuf_trap_t* register_trap(drakvuf_t drakvuf,
-                                  drakvuf_trap_info_t* info,
-                                  P* plugin,
+    virtual ~pluginex()
+    {};
+
+    // Params property is optional
+    template<typename Params = void, typename IB>
+    drakvuf_trap_t* register_trap(drakvuf_trap_info_t* info,
                                   hook_cb_t hook_cb,
                                   IB init_breakpoint,
-                                  const char* trap_name = nullptr,
-                                  AP ap = allocator<Params>(),
-                                  AT at = allocator<drakvuf_trap_t>())
+                                  const char* trap_name = nullptr)
     {
-        std::unique_ptr<Params> params;
-        std::unique_ptr<drakvuf_trap_t> trap;
-        if (!init_memory(plugin, trap, params, hook_cb, trap_name, ap, at))
-            return nullptr;
+        using Plugin = std::remove_pointer<decltype(this)>::type;
+
+        auto trap = new drakvuf_trap_t;
+
+        if constexpr (std::is_same_v<Params, void>)
+        {
+            trap->data = new plugin_data<Plugin, Params>(this, nullptr);
+        }
+        else
+        {
+            trap->data = new plugin_data<Plugin, Params>(this, new Params);
+        }
+
+        trap->name = trap_name;
+        trap->cb = hook_cb;
+        trap->type = BREAKPOINT;
 
-        if (!init_breakpoint(drakvuf, info, trap.get()))
+        if (!init_breakpoint(drakvuf, info, trap))
         {
             PRINT_DEBUG("%s for %s\n", ERROR_MSG_ADDING_TRAP, trap_name ? trap_name : trap->name);
             return nullptr;
```

Comment:
fix memory leak reported by scan-build (this return path leads to `trap` being leaked)
```suggestion
            delete trap;
            return nullptr;
```